content/docs/state-and-lifecycle.md

```diff
@@ -244,21 +244,21 @@ The `componentDidMount()` method runs after the component output has been render
   }
 ```
 
-Note how we save the timer ID right on `this`.
+لاحظ كيف حفظنا مُعرِّف عدّاد الوقت (timer ID) من خلال `this`.
 
-While `this.props` is set up by React itself and `this.state` has a special meaning, you are free to add additional fields to the class manually if you need to store something that doesn’t participate in the data flow (like a timer ID).
+لمّا كانت `this.props` يجري إعدادها عن طريق React نفسها و `this.state` تمتلك معنًى خاصًّا، فأنت حر بأن تضيف حقول إضافيّة يدويًّا إلى الصّنف إن احتجت تخزين شيء ما لا يُشارِك في تدفّق البيانات (مثل مُعرِّف عداد الوقت).
```

Comment:
> Please review the translation of `` While `this.props` is set up by React itself ``, we can use a better expression could be used here, like `في حين أن React هي المسؤولة عن إعداد...` or `بما أن React تُعدّ this.props...`.


I'm with you that there are better expressions for translation, but in the guide are forced in the first version to be a similar translation to Wiki-Hasoub .

Then , Should create a larger glossary and more clearly for fix all bad expressions.

So, let's postpone the change of bad expressions when we finish the first release

Reply:
It's not about the glossary. Grammatically, the sentence doesn't feel correct. Also, it provides a weird meaning since we say: `While (as when, or in the meantime) props are set up by React, and state has a meaning`, like the meaning could be confusing if we translate "While" to لما كانت. Also, I'm not sure that ` لما كان الشيء يجري إعداده فعلنا فعلا` is the best way to say `فعلنا فعلا في حين إعداد الشيء` or `فعلنا فعلا بينما تم إعداد الشيء` or `بينما يتم إعداد الشيء فعلنا فعلا`.

Hsoub's translation contains some errors, it's a super translation that we sould stick to it, but we should be carful.